manager/scheduler/filter.go

```diff
@@ -128,6 +128,13 @@ func (f *PluginFilter) SetTask(t *api.Task) bool {
 // Check returns true if the task can be scheduled into the given node.
 // TODO(amitshukla): investigate storing Plugins as a map so it can be easily probed
 func (f *PluginFilter) Check(n *NodeInfo) bool {
+	if n.Description.Engine == nil {
```

Comment:
Let's change this to `if n.Description == nil || n.Description.Engine == nil`.

It looks like not checking for nil here was an oversight that would currently allow a misbehaving or malicious worker to crash the manager.